src/lakefs_spec/transaction.py

```diff
@@ -0,0 +1,144 @@
+    ) -> Placeholder[Commit]:
+        """
+        Create a commit on a branch in a repository with a commit message and attached metadata.
+        """
+
+        # bind all arguments to the client helper function, and then add it to the file-/callstack.
+        op = partial(
+            commit, repository=repository, branch=branch, message=message, metadata=metadata
+        )
+        p: Placeholder[Commit] = Placeholder()
+        self.files.append((op, p))
+        # return a placeholder for the commit.
+        return p
+
+    def complete(self, commit: bool = True) -> None:
+        """
+        Finish transaction: Unwind file+versioning op stack via
+         1. Committing or discarding in case of a file, and
+         2. Conducting versioning operations using the file system's client.
+
+         No operations happen and all files are discarded if `commit` is False,
+         which is the case e.g. if an exception happens in the context manager.
+        """
+        while self.files:
+            # fsspec base class calls `append` on the file, which means we
+            # have to pop from the left to preserve order.
+            f = self.files.popleft()
+            if isinstance(f, AbstractBufferedFile):
+                if commit:
+                    f.commit()
+                else:
+                    f.discard()
+            else:
+                # client helper + return value case.
+                op, retval = f
+                if commit:
+                    result = op(self.fs.client)
+                    # if the transaction member returns a placeholder,
+                    # fill it with the result of the client helper.
+                    if isinstance(retval, Placeholder):
+                        retval.set_value(result)
+
+        self.fs._intrans = False
+
+    def create_branch(self, repository: str, branch: str, source_branch: str) -> str:
+        op = partial(
+            ensure_branch, repository=repository, branch=branch, source_branch=source_branch
+        )
+        self.files.append((op, branch))
+        return branch
+
+    def merge(self, repository: str, source_ref: str, into: str) -> None:
+        """Merge a branch into another branch in a repository."""
+        op = partial(merge, repository=repository, source_ref=source_ref, target_branch=into)
+        self.files.append((op, None))
+        return None
+
+    def revert(self, repository: str, branch: str, parent_number: int = 1) -> None:
+        """Revert a previous commit on a branch."""
+        op = partial(revert, repository=repository, branch=branch, parent_number=parent_number)
+        self.files.append((op, None))
+        return None
+
+    def rev_parse(
+        self, repository: str, ref: str | Placeholder[Commit], parent: int = 0
+    ) -> Placeholder[Commit]:
+        def rev_parse_op(client: LakeFSClient, **kwargs: Any) -> Commit:
```

Comment:
```suggestion
    ) -> Placeholder[Commit]:
    """Reverse parse the history of the ref."""
        def rev_parse_op(client: LakeFSClient, **kwargs: Any) -> Commit:
```

Reply:
"Rev" stands for revision, and only looks at that ref(erence) and not its history.